contracts/src/AgentExecutor.sol

```diff
@@ -36,4 +50,25 @@ contract AgentExecutor {
     function _transferToken(address token, address recipient, uint128 amount) internal {
         IERC20(token).safeTransfer(recipient, amount);
     }
+
+    /// @dev Register native asset from polkadto as ERC20 `token`.
+    function _registerToken(bytes32 agentID, bytes32 tokenID, string memory name, string memory symbol, uint8 decimals)
+        internal
+    {
+        IERC20 token = new ERC20(name, symbol, decimals);
+        Gateway(msg.sender).registerForeignToken(tokenID, address(token), agentID);
+    }
+
+    /// @dev Mint ERC20 token to `recipient`.
+    function _mintToken(bytes32 tokenID, address recipient, uint256 amount) internal {
+        address token = Gateway(msg.sender).tokenAddressOf(tokenID);
+        ERC20(token).mint(recipient, amount);
+        emit TokenMinted(tokenID, token, recipient, amount);
```

Comment:
Moving the events to ERC20 makes AgentExecutor simple again in terms of dependencies. It only depends on IERC20 interface.

Reply:
There is already an low level `Transfer` event in ERC20, the outside event could provide some more context like `tokenID`.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/79c1ba0f8848df15b15f3642c299e6b336cf2f81